### PR TITLE
Adds authTest

### PR DESCRIPTION
**Pull request by bwalsh**
This PR:
* adds a new test that runs badger with basic auth.
  * ensures that unauthorized user fails
  * ensures all tests pass with credentialed session

### CODE REVIEW

**Comment by bwalsh** [comment]
Still a work in progress.   Added simple basic auth config and make target to run tests in authentication mode.

**Comment by kellrott** [comment]
Target `develop`, that is the branch with the new auth system.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -200,3 +200,26 @@ jobs:
...
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v2
+    - name: Python Dependencies for Conformance
+      run: pip install requests numpy
+    - name: Download grip
+      uses: actions/download-artifact@v2
+      with:
+        name: gripBin
+    - name: Badger Test
```
Comment (kellrott) [inline .github/workflows/tests.yml:217]: Rename to something like Auth Test

File: Makefile [hunk 2]
```diff
@@ -119,6 +119,9 @@ test:
 test-conformance:
 	python conformance/run_conformance.py http://localhost:18201
 
+test-authorization:
+	python conformance/run_conformance.py http://localhost:18201 -u testuser -p abc123
```
Comment (kellrott) [inline Makefile:123]: Auth is going to have a lot of different pass/failure states then regular conformance. Maybe it should be written as a separate script.

File: test/badger-auth.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Default: badger
+
+Drivers:
+  badger:
+    Badger: grip-badger.db
+
+Server:
+  BasicAuth:
```
Comment (kellrott) [inline test/badger-auth.yml:8]: This config is probably going to be slightly different in develop branch

File: .github/workflows/tests.yml (hunk 1 above)
Comment (bwalsh) [reply .github/workflows/tests.yml:217]: done

**Pull request closed without merging**